nav2_ros_common/include/nav2_ros_common/lifecycle_node.hpp

```diff
@@ -395,6 +395,24 @@ class LifecycleNode : public rclcpp_lifecycle::LifecycleNode
   std::unique_ptr<bond::Bond> bond_{nullptr};
   double bond_heartbeat_period{0.1};
   rclcpp::TimerBase::SharedPtr autostart_timer_;
+
+private:
+  /**
+   * @brief Get the enable_lifecycle_services parameter value from NodeOptions
+   * @param options NodeOptions to check for the parameter
+   * @return true if lifecycle services should be enabled, false otherwise
+   */
+  static bool getEnableLifecycleServices(const rclcpp::NodeOptions & options)
+  {
+    // Check if the parameter is explicitly set in NodeOptions
+    for (const auto & param : options.parameter_overrides()) {
+      if (param.get_name() == "enable_lifecycle_services") {
+        return param.as_bool();
+      }
+    }
+    // Default to true if not specified
+    return true;
```

Comment:
The 2nd alternative, using `declare_parameter_if_not_declared` is:
```suggestion
    // Create a temporary node to properly declare and get the parameter
    auto temp_node = rclcpp::Node::make_shared("temp_lifecycle_param_node", options);
    
    // Use nav2 utilities to declare and get the parameter
    nav2_util::declare_parameter_if_not_declared(
      temp_node, "enable_lifecycle_services", rclcpp::ParameterValue(true));
    
    bool enable_lifecycle_services = true;
    temp_node->get_parameter("enable_lifecycle_services", enable_lifecycle_services);
    
    return enable_lifecycle_services;
```
CONS:
This builds a new temporary node, while the previous solution don't. 
PROS:
Allows to set param in YAML instead of via explicit C++ API (adding a param override to rclcpp::NodeOptions)

*Problems I see*:
the new temporary node gets created for each and every Lifecycle node, and with the same name. Can it cause troubles? Is it something we do not want to happen?